string location
string user
bool   cancel
---
int32 result
